### DISCUSSION TITLE
Securing the published AIR applications

### DISCUSSION THREAD

**Discussion by reibruzia** [airsdk/Adobe-Runtime-Support#2104] [General]
Hi,

I'm looking for some advice on how to secure the code and possibly the assets on a published application.

I've looked at SecureSWF and a few others but I'm  wondering maybe there's something better that I haven't heard of.

I've seen some use some sort of binary encryption where you can't even access the assets themselves on a decompiler like Trillix or JPEXS and I'm looking for something similar. 

Thank you

**Comment by 2jfw** [comment]
[**Related Feature Request here:** #494]

**Comment by ajwfrost** [comment]
Yes so we're adding a few things into our AIR SDK v50:

1) possibility of encrypting the whole SWF file. Basically this just encrypts the whole file - with the implication then for us needing to load in the whole file before we can decrypt and start using any of it - using a command-line tool provided within the AIR SDK. The big caveat here is that we do need to decrypt it internally, so if anyone is able to do a memory dump from the application process, they'll be able to find the SWF file within the memory if they know what they're looking for.

2) possibility of encrypting binary data that's embedded into a SWF. So e.g. using the `[Embed ...]` tag you can store any random file as binary data and later on you can access this as a byte array. (Have just checked this as I couldn't quite remember how it was done, I thought we were going to provide an encrypted string array.. which is what's mentioned in https://github.com/airsdk/Adobe-Runtime-Support/issues/494#issuecomment-995998347 - this doesn't appear to have been done yet although I think most of the underlying work is there...).

Just got one or two more issues to solve before pushing out the AIR SDK 50 beta release..

thanks
